### DISCUSSION TITLE
ShardingSphere and OceanBase project cooperation.

### DISCUSSION THREAD

**Discussion by dongzl** [oceanbase/oceanbase#1081] [Ideas]
Hi OceanBase community,

I want do some work for [`ShardingSphere`](https://github.com/apache/shardingsphere) and [`OceanBase`](https://github.com/oceanbase/oceanbase) project cooperation.

`ShardingSphere` is apache project, the ecosystem to transform any database into a distributed database system, and enhance it with sharding, elastic scaling, encryption features & more.

`ShardingSphere` provide these core feature:

- Data Sharding
- Distributed Transaction
- Read/Write Splitting
- HA
- Data Migration
- Query Federation
- Data Encryption
- Shadow Database
- DistSQL

`OceanBase` is a distributed relational SQL database built from scratch by Ant Group. It is highly scalable, resilient and can take on both transactional and analytical workloads, and it is highly compatible with MySQL syntax and features. 

If I want to some work for `ShardingSphere` and `OceanBase` cooperation.

- What are the main works and goals? 
- How to promote the project cooperation？

If someone has good advice, welcome to leave a message for discussion.

**Comment by whhe** [comment]
Hi @dongzl 

Welcome and thanks for your proposal. 

I'm not familar with ShardingSphere, seems that it supports MySQL for now, but I don't know whether that means it supports OceanBase as well. Maybe we can do the work to add OceanBase to the supported list firstly?

BTW, if you want to cooperate with the OceanBase community, you can disscuss with the others at GitHub like this thread or join our DingTalk group. You can also reach our developers directly in the SIG mail list, see: https://open.oceanbase.com/community/organization.

**Reply by dongzl** [reply]
> Hi @dongzl
> 
> Welcome and thanks for your proposal.
> 
> I'm not familar with ShardingSphere, seems that it supports MySQL for now, but I don't know whether that means it supports OceanBase as well. Maybe we can do the work to add OceanBase to the supported list firstly?
> 
> BTW, if you want to cooperate with the OceanBase community, you can disscuss with the others at GitHub like this thread or join our DingTalk group. You can also reach our developers directly in the SIG mail list, see: https://open.oceanbase.com/community/organization.

@whhe Thanks for your reply.  ShardingSphere has two products, `ShardingSphere-JDBC` and `ShardingSphere-Proxy`.

For `ShardingSphere-JDBC`, it is an enhancement to java JDBC. It provide `sharding` capability to RDB, such as `MySQL`, `PostgreSQL`, `SQLServer` and `Oracle`.It can make the database scale horizontally.

`ShardingSphere-JDBC` is only for java program. 

But `ShardingSphere-Proxy` is for other programming languages, It is a `Proxy` for `MySQL` and `PostgreSQL`, and implement database protocols. It also provide `sharding` capability to RDB.

@whhe  we can discuss this problem with Ding-talking, My account is `13436670847`.
